### PR TITLE
Issue #464 Fix: Added check for NOASSERTION 

### PR DESCRIPTION
**Pull request by spatil00**
Issue #464 Fix for issue

### CODE REVIEW

File: spdx/parsers/rdfbuilders.py [hunk 1]
```diff
@@ -163,6 +164,8 @@ def create_entity(self, doc, value):
             return self.build_person(doc, value)
         elif self.org_re.match(value):
             return self.build_org(doc, value)
+        elif value == "NOASSERTION":            # Supplier name could be NOASSERTION SPDX 2.3
+            return  utils.NoAssert()
```
Comment (meretp) [inline spdx/parsers/rdfbuilders.py:168]: If you in include this check here, any field that is created through this method, so basically every entity (e.g. the creator of the document, any reviewer, any annotator) could also be set to `NoAssert` which is not valid in general.

**Review by meretp — CHANGES_REQUESTED** [review]
Thanks for your contribution, the current codebase is not that easy to understand and your fix basically does what it should (especially it would also fix the same problem with `package originator` which could also be `NOASSERTION`). But your current solution would also allow other fields to be `NOASSERTION` which is not valid per spec (see my comment below). 
Looking deeper in the code, I have seen that the case distinction to `NOASSERTION` is already made [here](https://github.com/spdx/tools-python/blob/d6932e2d295a53c619deffd949ec30622a248765/spdx/parsers/rdf.py#L578) (and [here](https://github.com/spdx/tools-python/blob/d6932e2d295a53c619deffd949ec30622a248765/spdx/parsers/rdf.py#L564) for the package originator), just currently not seem to work properly. Could you try to fix the issue there?

**Comment by spatil00** [comment]
@meretp Thanks for comments , I have moved fix into jsonyamlxml.py , Its specific to supplier tag now, also I have taken liberty to fix same issue for package originator tag. Please let me know if this fix is fine can be merged.

**Comment by meretp** [comment]
Thanks for the effort, but I am sorry, maybe I wasn't clear on that, this issue is already fixed in jsonyamlxml.py. 
You could easily check this if you try to parse a Json document with supplier "NOASSERTION" on main. Your current fix would still fail for a RDF-file with package supplier being "NOASSERTION". 

You need to adapt the code in `spdx/parser/rdf.py` as I tried to point out. The following [logic](https://github.com/spdx/tools-python/blob/d6932e2d295a53c619deffd949ec30622a248765/spdx/parsers/rdf.py#L578) needs to be adapted:
```
 def p_pkg_suppl(self, p_term, predicate):
        for _s, _p, o in self.graph.triples((p_term, predicate, None)):
            try:
                if o == "NOASSERTION":
                    self.builder.set_pkg_supplier(self.doc, utils.NoAssert())
```
The return value `o` of `graph.triples()` is a `rdflib.Literal` and the comparison `o == "NOASSERTION"` is always False, so one would need to cast the type to `str` for this comparison. The same holds for `p_pkg_originator` some lines higher.

File: spdx/parsers/rdfbuilders.py [hunk 2]
```diff
@@ -162,7 +162,7 @@ def create_entity(self, doc, value):
         elif self.person_re.match(value):
             return self.build_person(doc, value)
         elif self.org_re.match(value):
-            return self.build_org(doc, value)
+            return self.build_org(doc, value)
```
Comment (DjDeepDive) [inline spdx/parsers/rdfbuilders.py:165]: No need to commit random whitespace, please

File: spdx/parsers/jsonyamlxml.py [hunk 3]
```diff
@@ -1264,8 +1265,9 @@ def parse_pkg_originator(self, pkg_originator):
         """
         if isinstance(pkg_originator, str):
             if pkg_originator == "NOASSERTION":
-                return self.builder.set_pkg_originator(self.document, NoAssert())
-            entity = self.builder.create_entity(self.document, pkg_originator)
+                return self.builder.set_pkg_originator(self.document,utils.NoAssert())
```
Comment (DjDeepDive) [inline spdx/parsers/jsonyamlxml.py:1268]: Stylistic check: Can you put a space between the comma and "utils" to be consistent on formatting?

**Comment by meretp** [comment]
@spatil00 we want to make a fix release soon where this issue should also be fixed. Do you plan to address the requested changes soon?

**Comment by spatil00** [comment]
Looks like the issue that I wanted to fix was already got fixed as mentioned by @meretp,

**Pull request closed without merging**